src/psyclone/tests/dynamo0p3_transformations_test.py

```diff
@@ -6474,3 +6475,546 @@ def test_no_acc():
+    '''Test that we can convert a synchronous halo exchange to an
+    asynchronous one using the Dynamo0p3AsyncHaloExchangeTrans
+    transformation and then move them to new valid locations. In this
+    case we move a haloexchangestart before a loop.
+
+    '''
+    _, invoke_info = parse(os.path.join(
+        os.path.dirname(os.path.abspath(__file__)),
+        "test_files", "dynamo0p3",
+        "4.5.2_multikernel_invokes.f90"),
+                           api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=True).create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    f2_hex = schedule.children[11]
+    ahex_trans = Dynamo0p3AsyncHaloExchangeTrans()
+    schedule, _ = ahex_trans.apply(f2_hex)
+
+    mtrans = MoveTrans()
+    schedule, _ = mtrans.apply(schedule.children[11],
+                               schedule.children[10])
+    result = str(psy.gen)
+    assert (
+        "      CALL f2_proxy%halo_exchange_start(depth=1)\n"
+        "      !\n"
+        "      DO cell=1,mesh%get_last_halo_cell(1)\n"
+        "        !\n"
+        "        CALL testkern_any_space_3_code(cell, nlayers, "
+        "op_proxy%ncell_3d, op_proxy%local_stencil, ndf_any_space_1_op, "
+        "ndf_any_space_2_op)\n"
+        "      END DO \n"
+        "      CALL f2_proxy%halo_exchange_finish(depth=1)\n") in result
+
+    if TEST_COMPILE:
+        # If compilation testing has been enabled (--compile flag
+        # to py.test)
+        assert code_compiles("dynamo0.3", psy, tmpdir, f90, f90flags)
+
+
+def test_async_hex_move_error_1():
+    '''Test that an asynchronous halo exchange start can not be moved
+    after its end and its end cannot be moved before its start
+
+    '''
+    _, invoke_info = parse(os.path.join(
+        os.path.dirname(os.path.abspath(__file__)),
+        "test_files", "dynamo0p3",
+        "1_single_invoke.f90"),
+                           api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=True).create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+
+    m1_hex = schedule.children[1]
+    ahex_trans = Dynamo0p3AsyncHaloExchangeTrans()
+    schedule, _ = ahex_trans.apply(m1_hex)
+
+    mtrans = MoveTrans()
+
+    # end before start
+    with pytest.raises(TransformationError) as excinfo:
+        schedule, _ = mtrans.apply(schedule.children[2],
+                                   schedule.children[1])
+    assert "dependencies forbid" in str(excinfo.value)
+
+    # start after end
+    with pytest.raises(TransformationError) as excinfo:
+        schedule, _ = mtrans.apply(schedule.children[1],
+                                   schedule.children[3])
+    assert "dependencies forbid" in str(excinfo.value)
+
+
+def test_async_hex_move_error_2():
+    '''Test that an asynchronous halo exchange start can not be moved
+    before a kernel that modifies the field and its end cannot be
+    moved after a kernel that reads the field
```

Comment:
Full stop please.

Reply:
Right, I've added missing full stops to all test descriptions. There were quite a few needed doing.